### PR TITLE
docs(samples): add samples for the Retail search service

### PR DESCRIPTION
**Pull request by t-karasova**

### CODE REVIEW

File: samples/search/update_attribute_configuration.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    # Update product
+    updated_product = get_product_service_client().update_product(
+        get_update_product_request(product_to_update))
+
+    print('---updated product---:')
+    print(updated_product)
+
+    print('---Wait 2 minutes to be sure the catalog has been indexed after the changes---:')
+    time.sleep(120)
```
Comment (kweinmeister) [inline samples/search/update_attribute_configuration.py:75]: Can you show how to check the response status for completion, vs waiting for a certain period of time?

File: samples/search/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+1. To create the bucket and upload the JSON file run the following command in the Terminal:
+
+    ```bash
+    python product/create_gcs_bucket.py
+    ```
+
+    Now you can see the bucket is created in the [Cloud Storage](https://console.cloud.google.com/storage/browser), and the files are uploaded.
+
+1. The name of the created GRS bucket is printed in the Terminal. Copy the name and set it as the environment variable `BUCKET_NAME`:
```
Comment (kweinmeister) [inline samples/search/README.md:85]: GCS bucket?

File: samples/search/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+To run a code sample from the Cloud Shell, you need to authenticate. To do this, use the Application Default Credentials.
+
+1. Set your user credentials to authenticate your requests to the Retail API
+
+    ```bash
+    gcloud auth application-default login
+    ```
+
+1. Type `Y` and press **Enter**. Click the link in Terminal. A browser window should appear asking you to log in using your Gmail account.
```
Comment (kweinmeister) [inline samples/search/README.md:30]: Is it your Gmail account or Google account?

File: samples/search/search_simple_query_test.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    assert re.match('.*results.*id.*', output)
+
+
+def test_search_simple_query_response():
+    response = search()
+
+    assert len(response.results) == 10
+    product_title = response.results[0].product.title
+    assert re.match('.*Hoodie.*', product_title)
+    #assert response.total_size == 51
```
Comment (kweinmeister) [inline samples/search/search_simple_query_test.py:39]: Update or remove this commented assertion

File: samples/search/search_with_pagination.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    offset = 0
+    page_token = ""
+
+    search_request_first_page = get_search_request("Hoodie", page_size, offset, page_token)
+    search_response_first_page = get_search_service_client().search(search_request_first_page)
+
+    print("---search response---")
+    print(search_response_first_page)
+
+    # PASTE CALL WITH NEXT PAGE TOKEN HERE:
```
Comment (kweinmeister) [inline samples/search/search_with_pagination.py:65]: Is there something supposed to be here in the PASTE CALL comments?

**Review by kweinmeister — CHANGES_REQUESTED** [review]

File: samples/search/update_attribute_configuration.py (hunk 1 above)
Comment (t-karasova) [reply samples/search/update_attribute_configuration.py:75]: In the tutorial we are asking user to update the product in catalog (make it searchable) and then to perform the search by this searchable attribute. So this message and the waiter is to make sure that the updated product will be indexed in catalog before user will search for it.
I can remove the waiter, but leave the message to let user know that there is an expected delay

File: samples/search/search_simple_query_test.py (hunk 4 above)
Comment (t-karasova) [reply samples/search/search_simple_query_test.py]: removed

File: samples/search/search_with_pagination.py (hunk 5 above)
Comment (t-karasova) [reply samples/search/search_with_pagination.py:65]: there is a tutorial where we are asking user to paste the next pease of code bellow this comment. But the code sample can be run without this additional code and the response should be valid.

File: samples/search/README.md (hunk 3 above)
Comment (t-karasova) [reply samples/search/README.md]: "Gmail" replaced with "Google"

File: samples/search/README.md (hunk 2 above)
Comment (t-karasova) [reply samples/search/README.md]: GRS replaced with GCS

File: samples/search/update_attribute_configuration.py (hunk 1 above)
Comment (t-karasova) [reply samples/search/update_attribute_configuration.py]: the waiter is removed

File: .kokoro/samples/lint/common.cfg [hunk 6]
```diff
@@ -31,4 +31,17 @@ gfile_resources: "/bigstore/cloud-devrel-kokoro-resources/python-docs-samples"
...
+    value: "945579214386"
+}
+env_vars: {
+    key: "BUCKET_NAME"
+    value: "crs-interactive-tutorials"
+}
+env_vars: {
+    key: "EVENTS_BUCKET_NAME"
+    value: "crs-interactive-tutorials"
+}
```
Comment (chingor13) [inline .kokoro/samples/lint/common.cfg:47]: From our meeting, we created the `retail-samples-secrets` secret so we can set the `SECRET_MANAGER_KEYS` env to `retail-samples-secrets`

Comment (t-karasova) [reply .kokoro/samples/lint/common.cfg]: I have added the SECRET_MANAGER_KEYS=retail-samples-secrets to the presubmit/common.cfg

File: samples/snippets/events/setup/setup_cleanup.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+def get_user_event(visitor_id):
+    timestamp = Timestamp()
+    timestamp.seconds = int(datetime.datetime.now().timestamp())
+
+    product = Product()
+    product.id = 'test_id'
+
+    product_detail = ProductDetail()
+    product_detail.product = product
+    product_detail.quantity = 3
```
Comment (chenlei1216) [inline samples/snippets/events/setup/setup_cleanup.py:51]: nit: quantity should not be set for "detail-page-view" event.

File: samples/snippets/product/crud_product.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    print(delete_product_request)
+
+    get_product_service_client().delete_product(delete_product_request)
+
+    print("deleting product " + product_name)
+    print("---product was deleted:---")
+
+
+# call the methods
+create_product()
```
Comment (chenlei1216) [inline samples/snippets/product/crud_product.py:138]: Worrying that if anything goes wrong in between and the product is not deleted, the test will fail the next time CreateProduct is called. Should we always call delete_product() before this, or use a random ID instead?

File: samples/snippets/search/search_with_boost_spec.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    print("---search request---")
+    print(search_request)
+
+    return search_request
+
+
+# call the Retail Search:
+def search():
+    # TRY DIFFERENT CONDITIONS HERE:
+    condition = '(colorFamily: ANY("Blue"))'
```
Comment (chenlei1216) [inline samples/snippets/search/search_with_boost_spec.py:60]: colorFamilies

File: samples/snippets/search/search_with_filtering.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+    print("---search request:---")
+    print(search_request)
+
+    return search_request
+
+
+# call the Retail Search:
+def search():
+    # TRY DIFFERENT FILTER EXPRESSIONS HERE:
+    filter_ = '(colorFamily: ANY("Black"))'
```
Comment (chenlei1216) [inline samples/snippets/search/search_with_filtering.py:53]: colorFamilies

**Review by chenlei1216 — APPROVED** [review]

File: samples/snippets/events/setup/setup_cleanup.py (hunk 7 above)
Comment (t-karasova) [reply samples/snippets/events/setup/setup_cleanup.py:51]: done

File: samples/snippets/events/import_user_events_big_query.py [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+    print("---the operation was started:----")
+    print(big_query_operation.operation.name)
+
+    while not big_query_operation.done():
+        print("---please wait till operation is done---")
+        time.sleep(5)
+
+    print("---import user events operation is done---")
+    print("---number of successfully imported events---")
+    print(big_query_operation.metadata.success_count)
```
[the remark this answers is not in the document]
Comment (t-karasova) [reply samples/snippets/events/import_user_events_big_query.py:87]: done

File: samples/snippets/product/crud_product.py (hunk 8 above)
Comment (t-karasova) [reply samples/snippets/product/crud_product.py:138]: done

File: samples/snippets/search/search_with_boost_spec.py (hunk 9 above)
Comment (t-karasova) [reply samples/snippets/search/search_with_boost_spec.py:60]: done

File: samples/snippets/search/search_with_filtering.py (hunk 10 above)
Comment (t-karasova) [reply samples/snippets/search/search_with_filtering.py:53]: done

**Review by chenlei1216 — APPROVED** [review]

**Review by kweinmeister — APPROVED** [review]

**Comment by t-karasova** [comment]
Hi Anthonios @parthea
Could you please check this changes and advise on how to include the tests to the CI build run?

**Comment by t-karasova** [comment]
the pull request is closed as it becomes to big, will be splitted into smaller pieces

**Pull request closed without merging**